### PR TITLE
Chapter filter for logged out state

### PR DESCRIPTION
**Pull request by KarinaMendez2**
for: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1154

### CODE REVIEW

File: src/app/content/studyGuides/components/Filters.spec.tsx [hunk 1]
```diff
@@ -49,4 +50,29 @@ describe('Filters', () => {
...
+      </MessageProvider>
+    </Provider>);
+
+    let message: string | undefined;
+    try {
+      component.root.findByType(ConnectedFilterList);
+    } catch (e) {
+      message = e.message;
+    }
+    expect(message).toEqual('No instances found with node type: "Connect(FiltersList)"');
```
Comment (PiotrKozlowski) [inline src/app/content/studyGuides/components/Filters.spec.tsx:75]: This works, however, it may be a problem if a message will change in the future due to some dependencies upgrade
We've been using `expect(() => component.root.findByType(ConnectedFilterList)).toThrow();` approach in some of the other cases which seems fine also for this one

**Review by PiotrKozlowski — DISMISSED** [review]

File: src/app/content/studyGuides/components/Filters.spec.tsx (hunk 1 above)
Comment (TomWoodward) [reply src/app/content/studyGuides/components/Filters.spec.tsx:75]: this strategy with the try/catch comparing the message is necessary for async logic, we do that in other places. in this case i think piotr's suggestion would be better

Comment (KarinaMendez2) [reply src/app/content/studyGuides/components/Filters.spec.tsx:75]: I tried that before, it throws the error anyways.

Comment (PiotrKozlowski) [reply src/app/content/studyGuides/components/Filters.spec.tsx:75]: @KarinaMendez2 Did you `expect` `() => component.root.findByType(ConnectedFilterList)` or only `component.root.findByType(ConnectedFilterList)` - the first will work, the second will throw an error

File: src/app/components/AllOrNone.tsx [hunk 2]
```diff
@@ -10,15 +12,19 @@ interface Props {
...
-  <FormattedMessage id='i18n:highlighting:filters:all'>
-    {(msg: Element | string) => <ButtonLink decorated onClick={onAll}>{msg}</ButtonLink>}
-  </FormattedMessage>
-  <span>|</span>
-  <FormattedMessage id='i18n:highlighting:filters:none'>
-    {(msg: Element | string) => <ButtonLink decorated onClick={onNone}>{msg}</ButtonLink>}
-  </FormattedMessage>
-</div>;
+const AllOrNone = ({className, onAll, onNone}: Props) => {
+  const userLoggedOut = useSelector(loggedOut);
```
Comment (TomWoodward) [inline src/app/components/AllOrNone.tsx:16]: this component is shared in places outside the scope of what you're trying to change, you can't add this logic here.

you could add a `disabled` prop, that would be ok

File: src/app/components/Button.tsx [hunk 3]
```diff
@@ -109,8 +109,12 @@ export const PlainButton = styled.button`
 // tslint:disable-next-line:variable-name
 export const ButtonLink = styled(PlainButton)`
   outline: none;
-  ${textStyle}
-  ${(props: {decorated: boolean}) => props.decorated ? decoratedLinkStyle : linkStyle}
+  ${(props: {decorated: boolean, disabled: boolean}) =>
+    props.disabled ? `
+      cursor: not-allowed;
```
Comment (TomWoodward) [inline src/app/components/Button.tsx:114]: are you sure you want to remove the link styling? is there design for this? i would think that you'd leave the existing styling and just add the disabled cursor.

you should do the cursor in a separate `${}` of the `ButtonLink` styles, there is no reason to make this logic so crazy.

File: src/app/components/Button.tsx [hunk 4]
```diff
@@ -109,8 +109,12 @@ export const PlainButton = styled.button`
...
+    `
+    : props.decorated ? {textStyle, decoratedLinkStyle} : {textStyle, linkStyle}
```
Comment (TomWoodward) [inline src/app/components/Button.tsx:116]: is this valid? i don't think this is valid

File: src/app/components/Checkbox.tsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ const CustomCheckbox = styled(
   }
 
   input:checked + & {
-    background-color: ${theme.color.primary.orange.darkest};
+    background-color: ${(props) => props.disabled ? theme.color.primary.gray.base : theme.color.primary.orange.darkest};
```
Comment (TomWoodward) [inline src/app/components/Checkbox.tsx:35]: don't you want to set the disabled style regardless of whether its checked or not?

File: src/app/content/components/popUp/ChapterFilter.tsx [hunk 6]
```diff
@@ -67,6 +69,8 @@ const ChapterFilter = ({
   selectedLocationFilters,
   setFilters,
 }: Props) => {
+  const userLoggedOut = useSelector(loggedOut);
```
Comment (TomWoodward) [inline src/app/content/components/popUp/ChapterFilter.tsx:72]: you can't add this auth logic here

File: src/app/content/studyGuides/components/Filters.tsx [hunk 7]
```diff
@@ -31,13 +32,16 @@ const ConnectedFilterList = connect(
   })
 )(FiltersList);
 
-export default () =>
-  <Filters>
+export default () => {
+  const userLoggedOut = useSelector(loggedOut);
```
Comment (TomWoodward) [inline src/app/content/studyGuides/components/Filters.tsx:36]: this is the correct place for the auth logic, from here just pass `disabled` props down where they are needed

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/app/components/Button.tsx (hunk 4 above)
Comment (KarinaMendez2) [reply src/app/components/Button.tsx:116]: it worked haha

File: src/app/components/Checkbox.tsx (hunk 5 above)
Comment (KarinaMendez2) [reply src/app/components/Checkbox.tsx:35]: no, because this sets the background of the checked box gray when disabled, which is only if it's in logged out state, that's what's in the design

Comment (TomWoodward) [reply src/app/components/Checkbox.tsx:35]: yea for the background, but i think the border needs to be changed when disabled as well

File: src/app/components/AllOrNone.tsx [hunk 8]
```diff
@@ -7,18 +7,21 @@ interface Props {
...
-  <FormattedMessage id='i18n:highlighting:filters:all'>
-    {(msg: Element | string) => <ButtonLink decorated onClick={onAll}>{msg}</ButtonLink>}
-  </FormattedMessage>
-  <span>|</span>
-  <FormattedMessage id='i18n:highlighting:filters:none'>
-    {(msg: Element | string) => <ButtonLink decorated onClick={onNone}>{msg}</ButtonLink>}
-  </FormattedMessage>
-</div>;
+const AllOrNone = ({className, onAll, onNone, disabled}: Props) => {
+  return <div className={className}>
```
Comment (TomWoodward) [inline src/app/components/AllOrNone.tsx:15]: why did you change this?

File: src/app/components/Checkbox.tsx [hunk 9]
```diff
@@ -18,7 +18,7 @@ interface Props {
 }
 
 // tslint:disable-next-line:variable-name
-const CustomCheckbox = styled(
+export const CustomCheckbox = styled(
```
Comment (TomWoodward) [inline src/app/components/Checkbox.tsx:21]: why?

File: src/app/components/Typography/index.ts [hunk 10]
```diff
@@ -52,3 +53,16 @@ export const labelStyle = css`
   line-height: 1.6rem;
   font-weight: normal;
 `;
+
+export const disabledStyle = css`
```
Comment (TomWoodward) [inline src/app/components/Typography/index.ts:57]: can you add this to `linkStyle` and `decoratedLinkStyle` instead of making a new thing? i'm pretty sure those css blocks can have functions in them that look at props

File: src/app/content/studyGuides/components/Filters.tsx [hunk 11]
```diff
@@ -9,7 +10,7 @@ import { setSummaryFilters } from '../actions';
 import * as selectors from '../selectors';
 
 // tslint:disable-next-line:variable-name
-const ConnectedChapterFilter = connect(
+export const ConnectedChapterFilter = connect(
```
Comment (TomWoodward) [inline src/app/content/studyGuides/components/Filters.tsx:13]: are you using this?

File: src/app/content/studyGuides/components/Filters.tsx [hunk 12]
```diff
@@ -21,7 +22,7 @@ const ConnectedChapterFilter = connect(
 )(ChapterFilter);
 
 // tslint:disable-next-line:variable-name
-const ConnectedFilterList = connect(
+export const ConnectedFilterList = connect(
```
Comment (TomWoodward) [inline src/app/content/studyGuides/components/Filters.tsx:25]: or this?

**Review by TomWoodward — CHANGES_REQUESTED** [review]

**Review by TomWoodward — APPROVED** [review]
